gpu/src/metal/fft/polynomial.rs

```diff
@@ -0,0 +1,145 @@
+use crate::metal::abstractions::state::MetalState;
+use lambdaworks_math::{
+    field::{
+        element::FieldElement,
+        traits::{IsTwoAdicField, RootsConfig},
+    },
+    polynomial::Polynomial,
+};
+
+use super::{errors::FFTMetalError, helpers::log2, ops::*};
+
+trait MetalFFTPoly<F: IsTwoAdicField> {
```

Comment:
```suggestion
pub trait MetalFFTPoly<F: IsTwoAdicField> {
```

This trait should be public to expose these functions outside the `gpu` crate.